.github/workflows/mypy.yml

```diff
@@ -0,0 +1,29 @@
+name: Mypy
+
+on: [push, pull_request]
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10"]
```

Comment:
Should add 3.8 here too since we've seen future annotations that only break in 3.8 and some people develop on the RGS environment